File: server/router/app/article.go
package app

import (
	v1 "server/api/v1"
	"server/middleware"

	"github.com/gin-gonic/gin"
)

type ArticltRouter struct{}

// InitArticleRouter 初始化 article 路由信息
func (s *ArticltRouter) InitArticleRouter(Router *gin.RouterGroup) {
	articleRouter := Router.Group("article").Use(middleware.OperationRecord())
	articleRouterWithoutRecord := Router.Group("article")
	var articleApi = v1.ApiGroupApp.AppApiGroup.ArticleApi
	{
		articleRouter.POST("createArticle", articleApi.CreateArticle)             // 新建article
		articleRouter.DELETE("deleteArticle/:id", articleApi.DeleteArticle)       // 删除article
		articleRouter.DELETE("deleteArticleByIds", articleApi.DeleteArticleByIds) // 批量删除article
		articleRouter.PUT("updateArticle/:id", articleApi.UpdateArticle)          // 更新article
	}
	{
		articleRouterWithoutRecord.GET("findArticle/:id", articleApi.FindArticle)   // 根据ID获取article
		articleRouterWithoutRecord.GET("getArticleList", articleApi.GetArticleList) // 获取article列表
	}
}

File: web/src/api/article.js
import service from '@/utils/request';

export const createArticle = (data) => {
  return service({
    url: '/article/createArticle',
    method: 'post',
    data,
  });
};

export const deleteArticle = (data) => {
  return service({
    url: '/article/deleteArticle',
    method: 'delete',
    data,
  });
};

export const deleteArticleByIds = (data) => {
  return service({
    url: '/article/deleteArticleByIds',
    method: 'delete',
    data,
  });
};

export const updateArticle = (data) => {
  return service({
    url: '/article/updateArticle',
    method: 'put',
    data,
  });
};

export const findArticle = (params) => {
  return service({
    url: '/article/findArticle',
    method: 'get',
    params,
  });
};

export const getArticleList = (params) => {
  return service({
    url: '/article/getArticleList',
    method: 'get',
    params,
  });
};
